travel/src/main/java/com/tx/travel/service/AuthService.java

```diff
@@ -25,55 +24,47 @@ public AuthService(final UserRepository userRepository, final RoleRepository rol
   }
 
   public void findByUsernameOrEmail(@NotNull final String username, @NotNull final String email)
-      throws UsernameAlreadyExistsException {
+      throws UsernameAlreadyExistsException,EmailAlreadyExistsException {
     final Optional<User> userByUsername = userRepository.findByUsername(username);
     if (userByUsername.isPresent()) {
       throw new UsernameAlreadyExistsException(username);
     }
     final Optional<User> userByEmail = userRepository.findByEmail(email);
     if (userByEmail.isPresent()) {
-      throw new UsernameAlreadyExistsException(email);
+      throw new EmailAlreadyExistsException(email);
     }
   }
 
-  public Set<Role> addRoles(final Set<String> strRoles) {
-    final Set<Role> roles = new HashSet<>();
-    if (strRoles == null) {
-      final Role userRole =
-          roleRepository
+  public Role addRoleAdmin(final String strRole) {
+    if (strRole == null) {
+      return roleRepository
+          .findByName(ERole.ROLE_EMPLOYEE)
+          .orElseThrow(() -> new RuntimeException(ERROR_ROLE_NOT_FOUND));
+    } else {
+      switch (strRole) {
+        case "admin" -> {
+          return roleRepository
+              .findByName(ERole.ROLE_ADMIN)
+              .orElseThrow(() -> new RuntimeException(ERROR_ROLE_NOT_FOUND));
+        }
+        case "mod" -> {
+          return roleRepository
+              .findByName(ERole.ROLE_OFFICE_MANAGER)
+              .orElseThrow(() -> new RuntimeException(ERROR_ROLE_NOT_FOUND));
+        }
+        default -> {
+          return roleRepository
               .findByName(ERole.ROLE_EMPLOYEE)
               .orElseThrow(() -> new RuntimeException(ERROR_ROLE_NOT_FOUND));
-      roles.add(userRole);
-    } else {
-      strRoles.forEach(
-          role -> {
-            switch (role) {
-              case "admin":
-                final Role adminRole =
-                    roleRepository
-                        .findByName(ERole.ROLE_ADMIN)
-                        .orElseThrow(() -> new RuntimeException(ERROR_ROLE_NOT_FOUND));
-                roles.add(adminRole);
-
-                break;
-              case "mod":
-                final Role modRole =
-                    roleRepository
-                        .findByName(ERole.ROLE_OFFICE_MANAGER)
-                        .orElseThrow(() -> new RuntimeException(ERROR_ROLE_NOT_FOUND));
-                roles.add(modRole);
-
-                break;
-              default:
-                final Role userRole =
-                    roleRepository
-                        .findByName(ERole.ROLE_EMPLOYEE)
-                        .orElseThrow(() -> new RuntimeException(ERROR_ROLE_NOT_FOUND));
-                roles.add(userRole);
-            }
-          });
+        }
+      }
     }
-    return roles;
+  }
+
+  public Role addRole(final ERole role) {
```

Comment:
here we have a method addRole and we are fetching the role, shouldn't it be `getRole` or?